src/memory/AmdFlash.hh

```diff
@@ -227,12 +228,25 @@ public:
 
 	enum class State { IDLE, IDENT, CFI, STATUS, PRGERR };
 
+	struct Sector {
+		const size_t address = 0;
+		const power_of_two<size_t> size = 1;
+		const bool writeProtect = false;
+		ptrdiff_t writeAddress = -1;
+		const uint8_t* readAddress = nullptr;
+
+		std::weak_ordering operator<=>(const Sector& sector) const { return address <=> sector.address; }
+		bool operator==(const Sector& sector) const = default;
+	};
```

Comment:
clang-tidy gives warnings about the 3 `const` members in this struct:
```
member 'address' of type 'const size_t' (aka 'const unsigned long') is const qualifiedclang-tidy(cppcoreguidelines-avoid-const-or-ref-data-members)
```
Last year I would also have made these members `const`, now I tend to agree with the guildines, although I still think it's unfortunate. See here for more details:
* https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/avoid-const-or-ref-data-members.html
* https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-constref

Summary: having `const` members makes the struct non-copy-assignable, and non-movable (it's still is copy-constructable). And when you put such objects in a vector, it's a useful optimization if you can move (instead of copy). (Though in this specific case move and copy are basically the same, but who knows what other members will be added in the future).

Reply:
Hmm yes unfortunate. I’ll change it.